forc-plugins/forc-client/src/op/call.rs

```diff
@@ -0,0 +1,2221 @@
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        let output_param = ParamType::try_from_type_application(&abi_func.output, &type_lookup)
+            .expect("Failed to convert output type");
+
+        let abi_encoder = ABIEncoder::new(EncoderConfig::default());
+        let encoded_data = abi_encoder.encode(&tokens)?;
+
+        let external_contracts = match external_contracts {
+            Some(external_contracts) => external_contracts,
+            None => {
+                // Automatically extract contract addresses from args; provide these as inputs to the call
+                // This makes the CLI more ergonomic
+                let extracted_external_contracts = extract_contract_addresses(&args);
+                if !extracted_external_contracts.is_empty() {
+                    forc_tracing::println_warning(
+                        "Automatically provided external contract addresses with call:",
+                    );
+                    extracted_external_contracts.iter().for_each(|addr| {
+                        forc_tracing::println_warning(&format!("- 0x{}", addr.to_string()));
+                    });
+                }
+                extracted_external_contracts
+            }
+        };
+
+        // Create and execute call
+        let call = ContractCall {
+            contract_id: contract_id.into(),
+            encoded_selector: encode_fn_selector(&selector),
+            encoded_args: Ok(encoded_data),
+            external_contracts: external_contracts
+                .iter()
+                .map(|addr| Bech32ContractId::from(*addr))
+                .collect(),
+            output_param: output_param.clone(),
+            is_payable: false,
+            custom_assets: Default::default(),
+        };
+
+        let provider = wallet.provider().unwrap();
+        // TODO: log decoding would be required for verbose debugging mode
+        let log_decoder = fuels::core::codec::LogDecoder::new(
+            fuels::core::codec::log_formatters_lookup(vec![], contract_id.into()),
+        );
+
+        let tx_status = match mode {
+            cmd::call::ExecutionMode::DryRun => {
+                let tx_policies = gas.map(Into::into).unwrap_or(TxPolicies::default());
+                let tx = call
+                    .build_tx(tx_policies, VariableOutputPolicy::default(), &wallet)
+                    .await
+                    .expect("Failed to build transaction");
+                provider
+                    .dry_run(tx)
+                    .await
+                    .expect("Failed to dry run transaction")
+            }
+            cmd::call::ExecutionMode::Simulate => {
+                forc_tracing::println_warning(&format!(
+                    "Simulating transaction with wallet... {}",
+                    wallet.address().hash()
+                ));
+                let tx_policies = gas.clone().map(Into::into).unwrap_or(TxPolicies::default());
```

Comment:
```suggestion
                let tx_policies = gas.as_ref().map(Into::into).unwrap_or_else(TxPolicies::default);
```

Can probably remove the clone by using as_ref